docs/config-file/v2.rst

```diff
@@ -636,6 +640,112 @@ Schema
 You can see the complete schema
 `here <https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/rtd_tests/fixtures/spec/v2/schema.yml>`_.
 
+.. _new-build:
+
+New ``build`` specification
+---------------------------
+
+.. warning::
+
+   This functionality is in beta. If you find any inconsistencies
+   or have feedback, please `open an
+   issue <https://github.com/readthedocs/readthedocs.org/issues/>`_.
+
+Read the Docs has introduced a more flexible ``build`` specification
+that allows you to use a more recent base image
+and control the versions of several tools:
+Python, Node.js, Rust, and Go.
+
+To opt in, specify the ``build.os`` setting
+and declare at least one tool version inside the ``build.tools`` key,
+as follows: 
+
+.. code-block:: yaml
+
+   version: 2
+
+   build:
+     os: ubuntu-20.04
+     tools:
+       python: 3.9
+       nodejs: 16
+       rust: 1.55
+       golang: 1.17
+
+The new ``build`` specification also supports
+the ``apt_packages`` key described above.
+
+.. warning::
+
+   The keys ``build.os`` and ``python.version`` cannot be used at the same time.
+   Doing so will error the build.
+
+build (new)
+~~~~~~~~~~~
+
+Configuration for the documentation build process.
+
+build.os
+````````
+
+The Docker image used for building the docs.
```

Comment:
It may be worth mentioning Read the Docs specifically in this phrase, like "The operating system the Read the Docs' based Docker image is based on". Otherwise, I think it can be interpreted as the regular `ubuntu:20.04` or any other image could be used.